Consolidate SubscriptionResource Usage

flvndvd opened the pull request:
## Description

<!-- Briefly describe the changes you've made and link any relevant issues (e.g., "Fixes #123"). -->
<!-- If the PR includes UI changes, please attach a screenshot or GIF to illustrate the modifications. -->
This PR consolidates usage of `SubscriptionResource` over `SubscriptionModel` in the Stripe webhook. The code could benefits of a refactoring as it's mainly repetitive code, but this is out-of-scope of this PR. The main goal here is to ensure `SubscriptionModel` is not used outside of the resource itself.

## Tests

<!-- Explain how you tested your changes, did you do it manually, did you add / update some existing tests ? See [here](https://www.notion.so/dust-tt/Guide-Testing-18428599d94180e09250ff256d6ac46e) -->

## Risk

Blast radius is pretty high as we are talking about `Subscription`. Safe to rollback.

## Deploy Plan

<!-- Outline the deployment steps. Specify the order of operations and any considerations that should be made before, during, and after deployment/ -->

flvndvd committed:
Consolidate SubscriptionResource Usage

adrsimon reviewed (approved):
lgtm!!

PopDaph commented on front/pages/api/stripe/webhook.ts:
```diff
@@ -1043,30 +1063,32 @@ async function handler(
-                status: "ended",
-                endDate: new Date(),
-              });
+              await matchingSubscription.markAsEnded();
```
This will put the subscription as "ended_backend_only" when we want it ended it since it comes from the webhook 🙏🏻  We need a parameter for `markAsEnded` 🙏🏻

Commit:
Take status param in markAsEnded

JulesBelveze committed:
[front] - feat(agent): public endpoint YAML export (#21501)

* [assistant/configuration] - feature: implement YAML export functionality for agent configurations

 - Created a new service to export agent configurations to a YAML file format
 - Refactored existing code to centralize YAML export logic into a new dedicated module
 - Updated the API endpoint to use the new YAML export service for downloading agent configurations
 - Removed duplicated and now redundant code from the previous implementation of YAML export in API endpoint

* [sdks] - feature: add schema for YAML export response type

 - Define a new schema to describe the response of exporting an agent configuration to YAML

[front] - fix: update API handler to use correct response type and method

 - Import the newly added `GetAgentConfigurationYAMLExportResponseType` to be used in Next API response type
 - Change the method to end the response correctly with `res.end` instead of `res.send` while serving yaml content

* [sdks/js] - refactor: simplify GetAgentConfigurationYAMLExportResponse schema

 - Changed the schema from an object to a plain string to represent YAML content directly
 - Removed unnecessary filename property from the schema as it may not be needed for YAML content representation

JulesBelveze committed:
[front] - refactor: streamline SkillBuilder form submission and attached knowledge updates (#21512)

- Removed condition to exclude skillId when creating a new skill, ensuring it is always sent if present
 - Centralized useFormContext hook utilization in SkillBuilderInstructionsEditor for form value setting
 - Removed redundant attachedKnowledge state and methods, integrating them directly into edit handlers
 - Optimized knowledge item collection logic into separate functions for better maintainability and readability
 - Enhanced debounced editor updates to interact with the useFormContext directly for setting form values

PopDaph reviewed (approved):
LGTM 🙏🏻
